olp-cpp-sdk-dataservice-read/src/generated/api/BlobApi.h

```diff
@@ -56,15 +57,16 @@ class BlobApi {
    * only supports a single byte range. The range parameter can also be
    * specified as a query parameter, i.e. range=bytes=10-. For volatile layers
    * use the pagination links returned in the response body.
-   * @param callback A callback function to invoke with the data response.
+   * @param context A CancellationContext, which can be used to cancel the pending request.
    *
    * @return The cancellation token.
    */
-
-  static client::CancellationToken GetBlob(
-      const client::OlpClient& client, const std::string& layerId,
-      const std::string& dataHandle, boost::optional<std::string> billingTag,
-      boost::optional<std::string> range, const DataResponseCallback& callback);
+  static DataResponse GetBlob(const client::OlpClient& client,
+                              const std::string& layer_id,
+                              const std::string& data_handle,
+                              boost::optional<std::string> billing_tag,
+                              boost::optional<std::string> range,
+                              client::CancellationContext context);
```

Comment:
I wonder if we really need to pass this by copy. Internally everything is sync and we only pass it to OlpClient which will take care that it copies it when necessary.

Reply:
I mean the `CancellationContext`, just to be clear. All other parameters should be ok by copy.